uint32 me
uint32 go_to_this_mode